File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/ISupportsEventBoxes.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Volo.Abp.EventBus.Distributed;

public interface ISupportsEventBoxes
{
    void PublishFromOutboxAsync(
        OutgoingEventInfo outgoingEvent,
        OutboxConfig outboxConfig
    );

    void PublishManyFromOutboxAsync(
        IEnumerable<OutgoingEventInfo> outgoingEvents,
        OutboxConfig outboxConfig
    );

    void ProcessFromInboxAsync(
        IncomingEventInfo incomingEvent,
        InboxConfig inboxConfig
    );
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/ISupportsEventBoxes.java
package com.ueueo.eventbus.distributed;

import java.util.List;

public interface ISupportsEventBoxes {
    void publishFromOutbox(
            OutgoingEventInfo outgoingEvent,
            OutboxConfig outboxConfig
    );

    void publishManyFromOutbox(
            List<OutgoingEventInfo> outgoingEvents,
            OutboxConfig outboxConfig
    );

    void processFromInbox(
            IncomingEventInfo incomingEvent,
            InboxConfig inboxConfig
    );
}
